provisioning/service/src/IQuery.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Provisioning.Service
+{
+    public interface IQuery : IDisposable
```

Comment:
We should avoid public interfaces where the API can change in the future.
If needed, this should instead be an abstract class that allows addition of new APIs.

Please add the reasoning on why we need this interface in the first place.

Reply:
Convert to an abstract class.

Reply:
done